src/main/java/com/google/sps/utils/data/Animal.java

```diff
@@ -0,0 +1,205 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collection;
+import java.util.HashSet;
+
+public class Animal {
+    
+    // Cross-referenced
+    private String common_name;
+    private String binomial_name;
+
+    // Provided by Wikipedia
+    private int population;
+    private String status;
+
+    // Provided by IUCN API
+    private int id;
+    private String kingdom;
+    private String phylum;
+    private String class;
+    private String order;
+    private String family;
+    private String genus;
+    private String population_trend;
+    private boolean marine_system;
+    private boolean freshwater_system;
+    private boolean terrestrial_system;
+    private String taxonomic_notes;
+    private Collection<String> countries;
+    private Collection<AnimalHistory> history_set;
+    private String citation_link;
+
+    public Animal() {
+        common_name = null;
+        binomial_name = null;
+        population = -1;
+        status = null;
+        id = -1;
+        kingdom = null;
+        phylum = null;
+        class = null;
+        order = null;
+        family = null;
+        genus = null;
+        population_trend = null;
+        marine_system = false;
+        freshwater_system = false;
+        terrestrial_system = false;
+        taxonomic_notes = null;
+        countries = new HashSet<>();
+        history_set = new HashSet<>();
+        citation_link = "https://apiv3.iucnredlist.org/api/v3/taxonredirect/";
+    }
+
```

Comment:
Could you add an additional Constructor that takes in common_name, binomial_name, population, status, trend, notes, and image_link as parameters? I think it'd be easier on my side to create a new Animal.

Reply:
Yeah that sounds good. Do you think we should both be able to create Animal objects using the data we collect (Wikipedia vs IUCN), or just have creation be from Wikipedia and then from the IUCN side just search by the name and add fields? If we can create the objects from both ends we should probably have a way of searching if the name already exists or something.

Reply:
Both ways should be good! It depends on if we still want to display information for an Animal without population information.

I think for searching, I was thinking of creating a Map with common_name being the key and then the Animal object being the value. Then we can just search if the common_name is already in the Map and then add parameters accordingly.